test.js

```diff
@@ -29,35 +29,86 @@ const format = async (format, code) => {
 
 // https://developer.mozilla.org/en-US/docs/Glossary/Void_element
 const allVoidElements = [
-  "area",
-  "base",
-  "br",
-  "col",
-  "embed",
-  "hr",
-  "img",
-  "input",
-  "link",
-  "meta",
-  "param",
-  "source",
-  "track",
-  "wbr",
+  { el: "area", hasTrailingNewline: false },
+  { el: "base", hasTrailingNewline: false },
+  { el: "br", hasTrailingNewline: false },
+  { el: "col", hasTrailingNewline: true },
+  { el: "embed", hasTrailingNewline: false },
+  { el: "hr", hasTrailingNewline: true },
+  { el: "img", hasTrailingNewline: false },
+  { el: "input", hasTrailingNewline: false },
+  { el: "link", hasTrailingNewline: false },
+  { el: "meta", hasTrailingNewline: false },
+  { el: "param", hasTrailingNewline: true },
+  { el: "source", hasTrailingNewline: true },
+  { el: "track", hasTrailingNewline: true },
+  { el: "wbr", hasTrailingNewline: false },
 ];
 
 allPrettierVersions.forEach((dep) => {
   const { default: prettier, version } = dep;
 
   test(`Prettier version ${version}`, async (t) => {
-    await t.test("preserve void syntax on all void elements", async () => {
+    await t.test("preserve void syntax on all void elements", async (t) => {
       const results = await Promise.all(
-        allVoidElements.map((el) => format(prettier.format, `<${el}>`)),
+        allVoidElements.map(({ el }) => format(prettier.format, `<${el}>`)),
       );
       results.forEach((formatted, index) => {
-        assert.equal(formatted, `<${allVoidElements[index]}>\n`);
+        assert.equal(formatted, `<${allVoidElements[index].el}>\n`);
       });
     });
 
+    await t.test(
+      "preserve void syntax on all void elements with following text",
+      async () => {
+        const results = await Promise.all(
+          allVoidElements.map(({ el }) =>
+            format(prettier.format, `<${el}>text`),
+          ),
+        );
+        results.forEach((formatted, index) => {
+          const { el, hasTrailingNewline } = allVoidElements[index];
+          assert.equal(
+            formatted,
+            `<${el}>${hasTrailingNewline ? "\n" : ""}text\n`,
+          );
+        });
+      },
+    );
+
+    await t.test(
+      "preserve void syntax on all void elements with following inline element",
+      async () => {
+        const results = await Promise.all(
+          allVoidElements.map(({ el }) =>
+            format(prettier.format, `<${el}><span></span>`),
+          ),
+        );
+        results.forEach((formatted, index) => {
+          const { el, hasTrailingNewline } = allVoidElements[index];
+          assert.equal(
+            formatted,
+            `<${el}>${hasTrailingNewline ? "\n" : ""}<span></span>\n`,
+          );
+        });
+      },
+    );
+
+    await t.test(
+      "preserve void syntax on all void elements with following block element",
```

Comment:
Can you please add one more test for void elements with following void element? The test string should be something like,

```ts
`<${el}><br>`
```